cmd/main.go

```diff
@@ -52,28 +53,44 @@ func main() {
 		command = reposerver.NewCommand()
 	case "argocd-cmp-server":
 		command = cmpserver.NewCommand()
-		isCLI = true
 	case "argocd-commit-server":
 		command = commitserver.NewCommand()
+		isArgocdCLI = true
 	case "argocd-dex":
 		command = dex.NewCommand()
 	case "argocd-notifications":
 		command = notification.NewCommand()
 	case "argocd-git-ask-pass":
 		command = gitaskpass.NewCommand()
-		isCLI = true
+		isArgocdCLI = true
 	case "argocd-applicationset-controller":
 		command = applicationset.NewCommand()
 	case "argocd-k8s-auth":
 		command = k8sauth.NewCommand()
-		isCLI = true
+		isArgocdCLI = true
 	default:
 		command = cli.NewCommand()
-		isCLI = true
+		isArgocdCLI = true
+	}
+	util.SetAutoMaxProcs(isArgocdCLI)
+
+	if isArgocdCLI {
+		// silence errors and usages since we'll be printing them manually.
+		// This is because if we execute a plugin, the initial
+		// errors and usage are always going to get printed that we don't want.
+		command.SilenceErrors = true
+		command.SilenceUsage = true
 	}
-	util.SetAutoMaxProcs(isCLI)
 
-	if err := command.Execute(); err != nil {
-		os.Exit(1)
+	err := command.Execute()
+	// if the err is non-nil, try to look for various scenarios
+	// such as if the error is from the execution of a normal argocd command,
+	// unknown command error or any other.
+	if err != nil {
+		pluginHandler := cli.NewDefaultPluginHandler([]string{"argocd"})
+		pluginErr := pluginHandler.HandleCommandExecutionError(err, isArgocdCLI, os.Args)
```

Comment:
The command execution error should probably receive the command object in parameter. This way, the handler can use functions such as `cmd.UsageString()` and `cmd.ErrPrefix()` to handle the behavior disabled by `SilenceUsage` and `SilenceErrors`.

Reply:
can you please take a look now?